rag/llm/model_meta.py

```
@@ -460,7 +461,131 @@ def _format_model_list(self, raw_model_list):
 
         return model_list
 
+class NVIDIA(OpenAIAPICompatible):
+    _FACTORY_NAME = "NVIDIA"
+    _HOSTED_API_HOST = "integrate.api.nvidia.com"
+    _CATALOG_URL = "https://api.ngc.nvidia.com/v2/search/catalog/resources/ENDPOINT"
+    _CATALOG_QUERY = {"page": 0, "pageSize": 500}
```

Ruff (0.16.0)

[warning] Mutable default value for class attribute

(RUF012)